Rubberduck.Core/UI/Inspections/InspectionResultsViewModel.cs

```diff
@@ -232,6 +234,106 @@ public bool GroupByLocation
+
+        public bool FilterInspectionsByHint
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Hint);
+            set
+            {
+                if (FilterInspectionsByHint == value) { return; }
+
+                if (value)
+                {
+                    _filterCriteria.Add(CodeInspectionSeverity.Hint);
+                }
+                else
+                {
+                    _filterCriteria.Remove(CodeInspectionSeverity.Hint);
+                }
+
+                FilterResults(_filterCriteria);
+                OnPropertyChanged();
+            }
+        }
+
+        public bool FilterInspectionsBySuggestion
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Suggestion);
+            set
+            {
+                if (FilterInspectionsBySuggestion == value) { return; }
+
+                if (value)
+                {
+                    _filterCriteria.Add(CodeInspectionSeverity.Suggestion);
+                }
+                else
+                {
+                    _filterCriteria.Remove(CodeInspectionSeverity.Suggestion);
+                }
+
+                FilterResults(_filterCriteria);
+                OnPropertyChanged();
+            }
+        }
+
+        public bool FilterInspectionsByWarning
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Warning);
+            set
+            {
+                if (FilterInspectionsByWarning == value) { return; }
+
+                if (value)
+                {
+                    _filterCriteria.Add(CodeInspectionSeverity.Warning);
+                }
+                else
+                {
+                    _filterCriteria.Remove(CodeInspectionSeverity.Warning);
+                }
+
+                FilterResults(_filterCriteria);
+                OnPropertyChanged();
+            }
+        }
+
+        public bool FilterInspectionsByError
+        {
+            get => _filterCriteria.Contains(CodeInspectionSeverity.Error);
+            set
+            {
+                if (FilterInspectionsByError == value) { return; }
+
+                if (value)
+                {
+                    _filterCriteria.Add(CodeInspectionSeverity.Error);
+                }
+                else
+                {
+                    _filterCriteria.Remove(CodeInspectionSeverity.Error);
+                }
+
+                FilterResults(_filterCriteria);
+                OnPropertyChanged();
+            }
+        }
+
+        public void FilterResults(IEnumerable<CodeInspectionSeverity> inspections)
+        {
+            if (_filterCriteria.Any())
+            {
+                Results = new ObservableCollection<IInspectionResult>(
+                        _resultsAll.Where(o => _filterCriteria.Contains(o.Inspection.Severity))
+                        .ToList());
+            }
```

Comment:
I'm confused. If `_filterCriteria` is being limited to a single element by the UI, why is it being maintained as a List? If the intention is to allow multiple criteria to be selected in the future, this is a much more efficient Linq expression that will scale more appropriately to large result lists:
```
_resultsAll.GroupBy(result => result.Inspection.Severity)
                    .Where(group => _filterCriteria.Contains(group.Key))
                    .SelectMany(x => x);
```